### PR TITLE
Prevent duplicate output commitments

### LINKED ISSUE #1497: duplicate commitments or duplicate outputs?
We currently check for "duplicate outputs" via - 

https://github.com/mimblewimble/grin/blob/da87f5fef4a3830c7c0f3dc8c7ec9d3414e63f44/chain/src/txhashset.rs#L661-L675

* look in the `output_pos` index for the commitment
* if we find an entry, lookup the hash at that pos
* compare the hash with the hash of the output

But I think this is not an exhaustive check and I _think_ maybe there is an issue here.

Note that we use the __commitment__ to lookup the pos in the MMR.
But we compare the hash of the __output__.

It is possible to have two different _outputs_ (with different hashes), both with the same _commitment_, specifically one could be a coinbase output and one a regular output (differing only in features).

So if we generated a (60 grin) coinbase output and reused the commitment for a (again 60 grin) regular output then we would have two different outputs, but they would share a non-unique commitment, even though they would hash to different values.

@ignopeverell - my understanding was that we wanted to disallow duplicate commitments in the output MMR. 
Is this correct? 
Or is it that we want to disallow duplicate outputs (which we do currently)?

### PR DESCRIPTION
**Pull request by antiochp**
Resolves #1497.

We were checking for duplicate _outputs_ (based on the hash in the MMR).
But we actually want to be checking for duplicate output _commitments_ directly.

TODO - 

- [x] Can we use a full archive node to check for any violations of this? (unlikely but will fork the chain if a duplicate is present).

### CODE REVIEW

File: chain/src/txhashset.rs [hunk 1]
```diff
@@ -659,16 +661,8 @@ impl<'a> Extension<'a> {
...
-			// as well as the index doesn't support rewind and is non-authoritative
-			// (non-historical node will have a much smaller one)
-			// note that this doesn't show the commitment *never* existed, just
-			// that this is not an existing unspent commitment right now
-			if let Some(hash) = self.output_pmmr.get_hash(pos) {
-				// Check the hash matches what we expect.
-				// We may be on a fork which may result in the entry at that pos being
-				// different to the one we expect.
-				if hash == OutputIdentifier::from_output(out).hash_with_index(pos - 1) {
+			if let Some(out_mmr) = self.output_pmmr.get_data(pos) {
```
Comment (ignopeverell) [inline chain/src/txhashset.rs]: As you mention this will hardfork archive nodes because `get_data` bypasses the spent check. We should only check UTXOs.

Comment (antiochp) [reply chain/src/txhashset.rs]: We are only checking UTXOs here.

The variants are `get_data()` and `get_data_from_file()`. 
`get_data()` _is_ "spent aware", the other goes straight to the file.

We should think about some better naming for these two...

**Comment by antiochp** [comment]
Kicked off a "full archival sync" to see if we successfully sync with this.
We should see a duplicate unspent commitment as a failure to sync fully.

Related - #1470 (noticing the early blocks are _slow_ to sync as not many nodes have them on the network).

File: chain/src/txhashset.rs (hunk 1 above)
Comment (ignopeverell) [reply chain/src/txhashset.rs]: Oh right, sorry for the confusion!

**Pull request merged**